app/models/invoice.rb

```diff
@@ -96,6 +96,8 @@ class Invoice < ApplicationRecord
   validates :issuing_date, :currency, presence: true
   validates :timezone, timezone: true, allow_nil: true
   validates :total_amount_cents, numericality: { greater_than_or_equal_to: 0 }
+  validates :payment_disputed, inclusion: { in: [false] }, if: :voided?
+  validates :payment_disputed, inclusion: { in: [false, true] }, unless: :voided?
```

Comment:
I think this second validation is unnecessary. First should be enough I think

Reply:
The second validation (as well as the first one) prevents `payment_disputed` to be valid when `nil`. 

```
invoice = FactoryBot.create(:invoice)
irb(main):014:0> invoice.payment_disputed = nil
=> nil
irb(main):015:0> invoice.valid?
=> false
```

Reply:
@ivannovosad I am not sure we usually add such validations regarding nil value, especially if this field is set internally. However, if you prefer to add it, I would rather use `exclusion: { in: [nil] }`

Reply:
I removed the `payment_disputed` completely, we only have `payment_dispute_lost_at` now